### PR TITLE
Added CL/CV Support and tests

### PR DESCRIPTION
**Pull request by flabbet**
I added a CL/CV (Travis CL, code Codecov) support and one test, that passes.

EDIT: Set up test enviroment with all enviroment variables. I am currently writing more tests

### CODE REVIEW

**Comment by flabbet** [comment]
Wrote 5 tests. Tomorrow I'll write some more

File: swaglyrics_backend/issue_maker.py [hunk 1]
```diff
@@ -444,6 +444,6 @@ def hello():
     return render_template('hello.html', unsupported_songs=data)
 
 
-if __name__ == '__main__':
+if not token and gh_token != '':
```
Comment (aadibajpai) [inline swaglyrics_backend/issue_maker.py]:
```suggestion
if not token and gh_token:
```
same difference, I'd also add a comment

**Comment by flabbet** [comment]
I covered most of main functions, except ones that use `@request_from_github()` decorator. I am just not successful with patching this. But I still think this is a good start

File: swaglyrics_backend/issue_maker.py [hunk 2]
```diff
@@ -50,10 +50,10 @@
 
 SQLALCHEMY_DATABASE_URI = "mysql+mysqlconnector://{username}:{password}@{username}.mysql.pythonanywhere-services." \
                           "com/{username}${databasename}".format(
-                                                                username=username,
```
Comment (aadibajpai) [inline swaglyrics_backend/issue_maker.py:56]: revert this maybe? formatting changes interfere with reviewing the main code

File: tests/test_issue_maker.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+import unittest
+from unittest.mock import patch
+from test.support import EnvironmentVarGuard
+
+from requests import Response
+
+
+class TestBase(unittest.TestCase):
+    def setUp(self):
+        self.env = EnvironmentVarGuard()
```
Comment (aadibajpai) [inline tests/test_issue_maker.py]: Tests look cool I'm just curious about this. Will we need to maintain all env vars like this? Maybe just os.environ could be patched class-wide?

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import swaglyrics_backend
+
+with open("README.md", "r") as fh:
+    long_description = fh.read()
+
+setuptools.setup(
+    name="swaglyrics_backend",
+    version=swaglyrics_backend.__version__,
+    author="Aadi Bajpai",
+    description="Server side code to make an issue on the main repo when the program encounters a song it can't fetch "
```
Comment (aadibajpai) [inline setup.py]: this can be a bit better 😛

File: tests/test_issue_maker.py (hunk 3 above)
Comment (flabbet) [reply tests/test_issue_maker.py]: Changed that

File: swaglyrics_backend/issue_maker.py (hunk 2 above)
Comment (aadibajpai) [reply swaglyrics_backend/issue_maker.py:56]: It's still slightly different :P

File: tests/test_issue_maker.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        response.status_code = 200
+        mock_get.return_value = response
+        mock_response.return_value = get_correct_spotify_search_json('sample_genius_data.json')
+        from swaglyrics_backend.issue_maker import genius_stripper
+        self.assertEqual(genius_stripper("Miracle", "Caravan Palace"), "Caravan-palace-miracle")
+
+    def test_that_title_mismatches(self):
+        from swaglyrics_backend.issue_maker import is_title_mismatched
+        self.assertTrue(
+            is_title_mismatched({"Bohemian", "Rhapsody", "by", "Queen"}, "Miracle by Caravan Palace", 2))
```
Comment (aadibajpai) [inline tests/test_issue_maker.py]: Why do you pass a set instead of a normal string?

File: tests/test_issue_maker.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+
+def get_correct_spotify_search_json(filename):
+    import flask
+    with open(filename, 'r') as r:
+        raw_json = r.read()
+        return flask.json.loads(raw_json)
+
+
+def generate_fake_unsupported():
```
Comment (aadibajpai) [inline tests/test_issue_maker.py:163]: this uses the same unsupported.txt or creates a new one in tests folder?

Comment (flabbet) [reply tests/test_issue_maker.py:163]: New one in tests folder

Comment (aadibajpai) [reply tests/test_issue_maker.py:163]: That's good

File: tests/test_issue_maker.py (hunk 5 above)
Comment (flabbet) [reply tests/test_issue_maker.py]: check `is_title_mismatched` function, it accepts words as first parameter and full title as second

Comment (aadibajpai) [reply tests/test_issue_maker.py]: Right so actually we call split() which returns a list. The thing with set is that every element should be unique which may not exist for some songs. You should use a list here to mimic as close to the actual functioning

Comment (aadibajpai) [reply tests/test_issue_maker.py]: https://github.com/SwagLyrics/swaglyrics-backend/blob/fda9ffe6345346d9da7c2dfb6dc895cc3cfdf7be/issue_maker.py#L148

Comment (flabbet) [reply tests/test_issue_maker.py]: Didn't know that set must have unique values, I'll change that

Comment (aadibajpai) [reply tests/test_issue_maker.py]: Yup.
image

**Comment by flabbet** [comment]
Fixed merge conflict. But somehow codecov broke. Logs look normal, idk why. I don't think it's my fault.

EDIT: It fixed itself somehow

**Pull request merged**